code/modules/reagents/reagent_containers/food/drinks/juicebox.dm

```diff
@@ -16,32 +68,63 @@
 	playsound(loc,'sound/effects/bonebreak1.ogg', rand(10,50), 1)
 	to_chat(user, SPAN_NOTICE("You pull off the straw and stab it into \the [src], perforating the foil!"))
-	icon_state = "[initial(icon_state)]_open"
 	atom_flags |= ATOM_FLAG_OPEN_CONTAINER
+	appearance = update_juicebox()
 
 /obj/item/chems/food/drinks/juicebox/apple
 	name = "apple juicebox"
 	desc = "A small cardboard juicebox with a cartoon apple on it."
-	icon_state = "juicebox_red"
 
 /obj/item/chems/food/drinks/juicebox/apple/Initialize()
 	. = ..()
+	appearance = SetColors("#ff0000", "#ffff00", "#ff0000", style="stripe")
 	reagents.add_reagent(/datum/reagent/drink/juice/apple, 25)
 
 /obj/item/chems/food/drinks/juicebox/orange
 	name = "orange juicebox"
 	desc = "A small cardboard juicebox with a cartoon orange on it."
-	icon_state = "juicebox_orange"
 
 /obj/item/chems/food/drinks/juicebox/orange/Initialize()
 	. = ..()
+	appearance = SetColors("#ffff00", "#ff0000", "#ffff00", style="stripe")
 	reagents.add_reagent(/datum/reagent/drink/juice/orange, 25)
 
 /obj/item/chems/food/drinks/juicebox/grape
 	name = "grape juicebox"
 	desc = "A small cardboard juicebox with some cartoon grapes on it."
-	icon_state = "juicebox_purple"
 
 /obj/item/chems/food/drinks/juicebox/grape/Initialize()
 	. = ..()
+	appearance = SetColors("#ff00ff", "#00ff00", style="stripe")
 	reagents.add_reagent(/datum/reagent/drink/juice/grape, 25)
+
+/obj/item/chems/food/drinks/juicebox/random/Initialize()
+	. = ..()
+	var/primary = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+	var/secondary = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+	var/logo = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+	appearance = SetColors(primary, pick(list(secondary, null)), pick(list(logo, null)), style=pick(list("basic", "stripe", "corner")))
+
+/obj/item/chems/food/drinks/juicebox/sensible_random
+	name = "Random Juicebox"
+	desc = "Juice in a box; who knows what flavor!"
+
+/obj/item/chems/food/drinks/juicebox/sensible_random/proc/juice_it()
+	var/datum/reagent/J = pick(subtypesof(/datum/reagent/drink/juice))
+	var/datum/reagent/K = pick(subtypesof(/datum/reagent/drink/juice) - J)
+	reagents.add_reagent(J, 20)
+	reagents.add_reagent(K, 5)
+	return reagents.reagent_list.Copy()
+
+/obj/item/chems/food/drinks/juicebox/sensible_random/Initialize()
+	. = ..()
+	var/list/juiceReagents = juice_it()
+	var/datum/reagent/J = juiceReagents[1]
+	var/datum/reagent/K = juiceReagents[2]
+	var/primary = J.color
+	var/secondary = rgb(rand(0, 255), rand(0, 255), rand(0, 255))
+	var/logo = K.color
+	appearance = SetColors(primary, secondary, logo, style=pick(list("stripe", "corner")))
+	var/splash = pick("teasing", "splash", "hint", "measure", "nip", "slug", "depth", "dash", "sensation", "surge", "squirt", "spritz", "efflux", "gush", "swell")
+	desc = "[J.name]; [J.description], this one comes with \an [splash] of [K.name] in a neat box."
+	name = "A juicebox of [J.name] and [K.name]"
```

Comment:
`"[J.name] and [K.name] juicebox"` would line up with the other drinks and not have the capitalized intial article.

Reply:
`Carrot Juice and Garlic Juice juicebox` is the example, the pronounciation of that bothers me some.